src/man/mod.rs

```diff
@@ -0,0 +1,373 @@
+use self::environment::Env;
+use self::flag::Flag;
+use self::option::Opt;
+use roff::{bold, italic, list, Roff, Troffable};
+use std::convert::AsRef;
+
+/// Man page struct.
+#[derive(Debug)]
+pub struct Man {
+  name: String,
+  description: Option<String>,
+  authors: Vec<Author>,
+  flags: Vec<Flag>,
+  options: Vec<Opt>,
+  environment: Vec<Env>,
+  arguments: Vec<String>,
+}
+
+impl Man {
+  /// Create a new instance.
+  pub fn new(name: &str) -> Self {
+    Self {
+      name: name.into(),
+      description: None,
+      authors: vec![],
+      flags: vec![],
+      options: vec![],
+      arguments: vec![],
+      environment: vec![],
+    }
+  }
+
+  /// Add a description.
+  pub fn description(mut self, desc: &str) -> Self {
+    let desc = desc.into();
+    self.description = Some(desc);
+    self
+  }
+
+  /// Add an author.
+  pub fn author(
+    mut self,
+    name: impl AsRef<str>,
+    email: Option<String>,
+  ) -> Self {
+    self.authors.push(Author {
+      name: name.as_ref().to_owned(),
+      email,
+    });
+    self
+  }
+
+  /// Add an environment variable.
+  pub fn environment(
+    mut self,
+    name: String,
+    default: Option<String>,
+    description: Option<String>,
+  ) -> Self {
+    self.environment.push(Env {
+      name,
+      default,
+      description,
+    });
+    self
+  }
+
+  /// Add an flag.
+  pub fn flag(
+    mut self,
+    short: Option<String>,
+    long: Option<String>,
+    description: Option<String>,
+  ) -> Self {
+    self.flags.push(Flag {
+      short,
+      long,
+      description,
+    });
+    self
+  }
+
+  /// Add an option.
+  pub fn option(
+    mut self,
+    short: Option<String>,
+    long: Option<String>,
+    description: Option<String>,
+    argument: String,
+    default: Option<String>,
+  ) -> Self {
+    self.options.push(Opt {
+      short,
+      long,
+      description,
+      argument,
+      default,
+    });
+    self
+  }
+
+  /// Add a positional argument. The items are displayed in the order they're
+  /// pushed.
+  // TODO: make this accept argument vecs / optional args too.  `arg...`, `arg?`
+  pub fn argument(mut self, arg: String) -> Self {
+    self.arguments.push(arg);
+    self
+  }
+
+  pub fn render(self) -> String {
+    let man_num = 1;
+    let mut page = Roff::new(&self.name, man_num);
+    page = description(page, &self.name, &self.description);
+    page = synopsis(
+      page,
+      &self.name,
+      &self.flags,
+      &self.options,
+      &self.arguments,
+    );
+    page = flags(page, &self.flags);
+    page = options(page, &self.options);
+    page = environment(page, &self.environment);
+    page = exit_status(page);
+    page = authors(page, &self.authors);
+    page.render()
+  }
+}
+
+/// Create a `NAME` section.
+///
+/// ## Formatting
+/// ```txt
+/// NAME
+///         mycmd - brief description of the application
+/// ```
+#[inline]
+pub fn description(page: Roff, name: &str, desc: &Option<String>) -> Roff {
```

Comment:
Why is this marked as inline? The compiler should be clever enough to inline this if wise.

And why is it public? It's only used in render, right?

Reply:
Same for all the fns below

Reply:
I was thinking of exposing them all as public, but I reckon you're right and we should just keep them private. Will change!